docs/proposals/PROPOSAL_F.md

```diff
@@ -0,0 +1,112 @@
+
+- **provenance**: a provenance tool can easily keep a database of image metadata based on artifacts alone to curate images available on a system, or similar use cases that require such a database.
+
+### Benefits
+
+- A compatibility interest group is free to namespace and define whatever metadata annotations their tooling needs. For example, if a graph schema is required to describe relationships between attributes, the reference URL can be a known metadata attribute. If documentation is desired, this is possible too.
+- The simple structure supports the most basic of use cases (key/value pair checking) to more advanced (graph-based scheduling).
+
+## Image Compatibility Artifact
+
+In [Proposal B](PROPOSAL_B.md) it is stated that a compatibility artifact should describe CRITICAL requirements for container images to run on a host, which is inferred to mean a limited set that determines compatibility. This proposal takes the stance that we don't yet know what that minimal set is, and further, that it may vary depending on the use case for any specific artifact. Instead of declaring a minimum set, this proposal states that the artifact should contain minimum requirements for compatibility but can contain more that MAY be used by plugins to not just determine compatibility but optimal usage. The artifact serves as a source of metadata about container requirements, and it is on the level of the plugins or associated tooling to decide which subset should be used for a given use case. Arguably, this set will change. More discussion and research is required to understand what the line between critical (for the image to run, compatibility) and running optimally.
+
+### Artifact Manifest
+
+This artifact MAY have a manifest that references it with the media type that is specific to the compatibility interest group. For example, metadata for node feature discovery may take the following form:
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.oci.image-compatibility.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/k8s.nfd.image-compatibility.spec.v1+json",
+      "digest": "sha256:4a47f8ae4c713906618413cb9795824d09eeadf948729e213a1ba11a1e31d052",
+      "size": 1710
+    }
+  ],
+  "annotations": {
+    "oci.opencontainers.image.created": "2024-01-02T03:04:05Z"
+  }
+}
+```
+
+### Artifact
+
+The artifact is a simple json structure of named groups and within them, key value pairs. The groups can provide a higher level organization, such as types of features or even versions. This organization is up to the compatibility interest group. Here is an example organizing NFD annotations within extractor groups (a random selection):
```

Comment:
>simple json structure of named groups and within them, key value pairs

and 

> the media type that is specific to the compatibility interest group

and

>This organization is up to the compatibility interest group

if a media type organization is up to compatibility interest group, does it mean except the artifiact type and media type definition, the content structure should be defined somewhere else

Reply:
The structure should be suggested by the working group here. The compatibility interest group defines the metadata within. We discussed this a bit in the call today.

Reply:
Can we just leave it to the same limitations as JSON? Only having key pairs will just mean more complex data will get jammed in the key, making it harder to parse.

Reply:
The structure has to be in json, yes.

Reply:
So any valid JSON, not just key pairs?

Reply:
I am pretty indifferent to that, whatever the working group decides is best.